File: Dished/View Controllers/DANavigationController.m
//
//  DANavigationController.m
//  Dished
//
//  Created by Ryan Khalili on 9/6/14.
//  Copyright (c) 2014 Dished. All rights reserved.
//

#import "DANavigationController.h"


@implementation DANavigationController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [self adjustAppearance];
}

- (void)adjustAppearance
{
    CGRect hairlineRect = CGRectMake( 0, self.navigationBar.frame.size.height, self.navigationBar.frame.size.width, 0.5 );
    UIView *navBorder = [[UIView alloc] initWithFrame:hairlineRect];
    
    [navBorder setBackgroundColor:[UIColor colorWithRed:0.78 green:0.78 blue:0.78 alpha:1]];
    [navBorder setOpaque:YES];
    [self.navigationBar addSubview:navBorder];
    
    self.navigationBar.barTintColor = [UIColor whiteColor];
}

@end

File: Dished/View Controllers/DANavigationController.swift
//
//  DANavigationController.swift
//  Dished
//
//  Created by Ryan Khalili on 3/21/15.
//  Copyright (c) 2015 Dished. All rights reserved.
//

import UIKit

class DANavigationController: UINavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()
        adjustAppearance()
    }
    
    private func adjustAppearance() {
        let barFrame = navigationBar.frame
        let hairlineRect = CGRectMake(0, barFrame.size.height, barFrame.size.width, 0.5)
        
        let navBorder = UIView(frame: hairlineRect)
        navBorder.backgroundColor = UIColor(red: 0.78, green: 0.78, blue: 0.78, alpha: 1.0)
        navBorder.opaque = true
        navigationBar.addSubview(navBorder)
        
        navigationBar.barTintColor = UIColor.whiteColor()
    }
}